examples/python-dataflow/object_detection.py

```diff
@@ -10,7 +10,12 @@
 
 model = YOLO("yolov8n.pt")
 
-node = Node()
+try:
+    node = Node()
+except RuntimeError as e:
+    print("Dataflow initialization failed: ", e)
+    exit(0)
```

Comment:
I don't think that it's a good idea to treat initialization errors as success. You might miss the error message in the logs and think that everything worked without issues.

For our CI, we want the job to fail in case of an error, otherwise it doesn't make sense to run the examples as integration tests.

Reply:
Okay, the problem that i'm trying to solve is to not have a wall of errors when only one node fails on initialization making it easy to debug.

How can we do this in a way that it's clear.

This solution will always return an error on the dataflow if one node fails, so this does not silently let a dataflow run.

Reply:
Want to emphasize that real estate in the terminal is expensive and I don´t think that having:
```
  2024-06-09T19:28:23.247953Z ERROR dora_daemon::spawn: 018ffe79-1c0c-7b68-8886-d8cff079c232/object_detection: 
Traceback (most recent call last):
  File "/home/peter/Documents/work/dora/examples/python-dataflow/object_detection.py", line 13, in <module>
    node = Node()
RuntimeError: failed to init event stream

Caused by:
    subscribe failed: Some nodes exited before subscribing to dora: {NodeId("plot")}

    This is typically happens when an initialization error occurs
                    in the node or operator. To check the output of the failed
                    nodes, run `dora logs 018ffe79-1c0c-7b68-8886-d8cff079c232 plot`.

Location:
    apis/rust/node/src/event_stream/mod.rs:90:17

    at binaries/daemon/src/spawn.rs:371
```

Helps anyone debug

Reply:
Yes, I fully agree with you on that! My concern is that "subscribe failed: Some nodes exited before subscribing to dora" is not the only reason why initialization can fail. For example, there could also be an error when trying to initialize the shared memory used for communication. If such an error happens with this implementation, the dataflow would finish with a success status code and completely hide the error.

Reply:
The proper solution is probably to not print all error messages to stdout. Instead, we should filter the observed error types and only display the most relevant. We still write all errors to the log files though.

I have some ideas how we could implement this. I try to draft something up in the next few days.